### ISSUE TITLE
Should we expose our style guide as a gem?

### ISSUE THREAD

**Issue by sambostock** [Shopify/ruby-style-guide#140]
# TL;DR
It might make sense for us to expose the styleguide as a gem, allowing configuration updates to become part of the predicatable Dependabot auto-update process.

----

# Details

As brought up [in Slack](https://shopify.slack.com/archives/C0DRQ0HEJ/p1581532645004500)

> Currently, the way we share our Rubocop config across repos is
> ```yml
> # .rubocop.yml
> inherit_from:
>   - http://shopify.github.io/ruby-style-guide/rubocop.yml
> ```
> The thing is, this doesn't follow any kind of versioning. It just follows the rules from [inheriting from a remote URL](https://rubocop.readthedocs.io/en/latest/configuration/#inheriting-configuration-from-a-remote-url)
>
>> The remote config file is cached locally and is only updated if:
>> - The file does not exist.
>> - The file has not been updated in the last 24 hours.
>> - The remote copy has a newer modification time than the local copy.
>
> Sometimes the file is gitignored, sometimes it isn't.
> - If it is ignored, then sometimes new rules start being enforced in CI.
> - If it isn't, then sometimes a random PR will have updates to the style config.
>
> **Given that we have Dependabot to keep dependencies up to date, and it is possible to include Rubocop config in a gem, would it make sense to publish our config as a gem, and use [inherit_gem](https://rubocop.readthedocs.io/en/latest/configuration/#inheriting-configuration-from-a-dependency-gem) to apply it?**

Other benefits that come to mind are:
- We could explicitly make Rubocop a dependency.
  _No more incompatible cops due to version mismatches!_
- We could use existing dependency tooling to check for up-to-date style configs (e.g. Services Internal classification checks)
- We could add dependencies on the extracted cop gems (e.g. `rubocop-performance`), without requiring the client to know to install those cops

If we wanted to get really bold, maybe a gem would even be a good place to put the common foundation for a command similar to `dev style --include-branch-commits`, instead of every repo implementing it on their own.

----

cc. @rafaelfranca
